server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1201,34 +1204,158 @@ void removeEntitlement(Entitlement entitlement,
         sink.queueEvent(event);
     }
 
+
+    /**
+     * Revokes the given set of entitlements.
+     *
+     * @param entsToRevoke entitlements to revoke
+     */
+    @Override
+    @Transactional
+    public void revokeEntitlements(List<Entitlement> entsToRevoke) {
+        if (entsToRevoke.isEmpty()) {
+            return;
+        }
+
+        log.info("Batch revoking entitlements: " + entsToRevoke.size());
+
+        entsToRevoke =  new ArrayList<Entitlement>(entsToRevoke);
+
+        List<Pool> poolsToDelete = poolCurator.listBySourceEntitlements(entsToRevoke);
+
+        for (Pool pool : poolsToDelete) {
+            entsToRevoke.addAll(pool.getEntitlements());
+        }
```

Comment:
You are refering to listBySourceEntitlements(X)? That method is not looking for 'sibling' entitlements of X. I think the idea has been (and I kept it) that the method looks for entitlements that have been created as a consequence of X. So for example bonus pools that have been created as a consequence of entitlements X.